### ISSUE TITLE
Add support for CustomResourceDefinitions (and phase out TPRs)

### ISSUE THREAD

**Issue by luxas** [heptio/sonobuoy#25]
As you all know, CRDs are the successor of TPRs. 
TPRs will be removed in v1.8, make sure sonobuoy can handle that gracefully ;)

**Comment by timothysc** [comment]
Yup, this is known.  We just wanted to get it out and I'll update soon-ish.

**Comment by timothysc** [comment]
I will also need to update the deps on this one.

**Comment by kensimon** [comment]
Thoughts:

CRDs aren't a part of client-go yet, you need to initialize a separate clientset for everything under apiextensions.  It seems like it will eventually be rolled in, see: https://github.com/kubernetes/client-go/issues/247

We can refactor pretty heavily to ensure the clientset and the apiextensions clientset are both passed around to the various discovery functions, or we can wait for the above issue to be addressed (and maybe help with it.)

**Comment by timothysc** [comment]
We should wait until CRDs are in client-go.  

Blarg, I want to avoid the vendor nightmare that we pulled before.  Perhaps we should poke @ncdc on this issue.

**Comment by ncdc** [comment]
@kensimon @timothysc I'm not sure what you mean about CRDs not being in client-go? Are you talking specifically about having a typed client to be able to register CRDs? If so, then you're correct, that's part of apiextensions. But we've stated that we'll ship CRDs as yaml and require admins to install them via kubectl. Or are you talking about something slightly different?

**Comment by timothysc** [comment]
@ncdc this is about scanning and reporting what is there for diagnostics and reporting.

**Comment by ncdc** [comment]
I see. I would vendor in what you need from apiextensions for now, or if that's too painful, just build your own client.

**Comment by timothysc** [comment]
This is pending an updated to client... on hold blocked atm.

**Comment by timothysc** [comment]
This is now unblocked, I'll have a fix soonish.

**Comment by timothysc** [comment]
This may be a push b/c client-go doesn't actually have the straight up interface that I'm looking for, but exists in the main repository.  

client.Apiextensions().CustomResourceDefinitions().List(options)  

which is essentially under the ApiextensionsV1beta1 group is nowhere to be found in the client, but I'm sure could be accessed through discovery.

**Comment by timothysc** [comment]
It requires a separate client util+wrapper k8s.io/apiextensions-apiserver which is outside of client-go.

**Comment by luxas** [comment]
Thanks for fixing this @timothysc!
